connectors/byoc.py

```diff
@@ -508,7 +508,9 @@ async def prepare(self, config):
                 )
         except Exception as e:
             logger.critical(e, exc_info=True)
-            raise DataSourceError(f"Could not instantiate {fqn} for {service_type}")
+            raise DataSourceError(
+                f"Could not instantiate {fqn} for {service_type}. Exception encountered: {e}"
+            )
```

Comment:
Will [this](https://stackoverflow.com/questions/1350671/inner-exception-with-traceback-in-python) work?

```suggestion
            raise DataSourceError(f"Could not instantiate {fqn} for {service_type}") from e
```

Reply:
Let me take a look, I'll update you.

Reply:
I checked, But still, with this suggestion, the message will not be visible in the KibanaUI.

Reply:
@jignesh-crest does it have to be visible in KibanaUI? That's a purely techincal error that should be accessed via logs of the connector. It does not seem to make sense to display it to the user - maybe worth to just say something like "Check service logs for the exception" in the message.

Moreover, the user should never see this error - it's seen only when connector is in development.

Reply:
Where would they look at these logs if it's a native connector @artem-shelkovnikov ? I would like to confirm that once this is a native connector that we could say: Click to Discover -> Logs and surface these errors more specifically

Reply:
I'm actually not 100% sure - I know it'll be "same place where you look at Enterprise Search logs". So if log shipping is enabled, it should be in different deployment, otherwise in same deployment in logs as you described.